.github/workflows/buildAndPublishReactNativeArtifacts.yml

```
@@ -0,0 +1,213 @@
+
+      - name: Check if NewDot patches or react-native version have changed
+        run: |
+          VERSION=$(jq -r '.dependencies["react-native"]' package.json)
+          HASH=$(VERSION=$VERSION ./.github/scripts/computePatchesHash.sh)
+
+          git fetch origin ${{ github.event.before }}
+          git checkout ${{ github.event.before }}
+
+          OLD_VERSION=$(jq -r '.dependencies["react-native"]' package.json)
+          # If version has changed and we want to find differences in patches, we need to consider pattern that involves previous react-native version
+          OLD_HASH=$(VERSION=$OLD_VERSION ./.github/scripts/computePatchesHash.sh)
+
+          if [ "$HASH" == "$OLD_HASH" ] && [ "$VERSION" == "$OLD_VERSION" ]; then
+            echo "No relevant changes"
+            exit 1
+          fi
+
+          if [ "$VERSION" != "$OLD_VERSION" ]; then
+            echo "Detected react-native version bump ($OLD_VERSION -> $VERSION)"
+            exit 0
+          fi
+
+          if [ "$HASH" != "$OLD_HASH" ]; then
+            echo "Detected changes in NewDot patches:"
+            git diff --name-only ${{ github.event.before }}..${{ github.event.after }} -- "patches/react-native+*.patch"
+          fi
+        env:
+          PATCHES_PATHS: "patches"
+
+  verifyHybridAppPatches:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        # v4
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608
+        with:
+            submodules: true
+            fetch-depth: 0
+      - name: Get react-native version
+        run: |
+          echo "VERSION=$(jq -r '.dependencies["react-native"]' package.json)" >> $GITHUB_ENV
+
+      - name: Check if HybridApp patches have changed
+        run: |
+          CURRENT_SUBMODULE_COMMIT=$(git rev-parse :Mobile-Expensify)
+          CURRENT_PATCHES_HASH=$(./.github/scripts/computePatchesHash.sh)
+ 
+          git fetch origin ${{ github.event.before }}
+          git checkout ${{ github.event.before }}
+
+          git submodule update
+          
+          PREVIOUS_SUBMODULE_COMMIT=$(git rev-parse :Mobile-Expensify)
+
+          if [ "$CURRENT_SUBMODULE_COMMIT" == "$PREVIOUS_SUBMODULE_COMMIT" ]; then
+            echo "Submodule commit is the same as the previous one, skipping patch check"
+            exit 1
+          fi
+
+          PREVIOUS_PATCHES_HASH=$(./.github/scripts/computePatchesHash.sh)
+
+          if [ "$CURRENT_PATCHES_HASH" == "$PREVIOUS_PATCHES_HASH" ]; then
+            echo "No changes in HybridApp patches"
+            exit 1
+          fi
+
+          echo "Detected changes in HybridApp patches:"
+          git -C Mobile-Expensify diff --name-only $PREVIOUS_SUBMODULE_COMMIT..$CURRENT_SUBMODULE_COMMIT -- "patches/react-native+*.patch"
+        env:
+          PATCHES_PATHS: "Mobile-Expensify/patches"
+          VERSION: ${{ env.VERSION }}
+
+  buildAndPublishRNArtifactsForStandaloneNewDot:
+    runs-on: ubuntu-latest
```

We can use large runners for the build, but only on Expensify org. Otherwise GitHub will start billing anyone for large runner minutes if they push code to main in their fork:

```suggestion
    runs-on: ${{ github.repository_owner == 'Expensify' && 'ubuntu-latest-xl' || 'ubuntu-latest' }}
```